File: samples/spring-mvc-ap/src/main/java/io/openapiprocessor/samples/EchoController.java
package io.openapiprocessor.samples;

import io.openapiprocessor.openapi.api.EchoApi;
import org.springframework.web.bind.annotation.RestController;

/**
 * Implementation of the echo api defined in src/api/openapi.yaml.
 */
@RestController
public class EchoController implements EchoApi {

    @Override
    public String getEcho (String source) {
        return source;
    }

}

File: samples/micronaut-4-reactive-kt/src/main/kotlin/io/openapiprocessor/samples/echo/server/EchoController.kt
package io.openapiprocessor.samples.echo.server

import io.micronaut.http.annotation.Controller
import io.openapiprocessor.openapi.api.EchoApi
import io.openapiprocessor.openapi.model.Echo
import jakarta.validation.Valid
import org.reactivestreams.Publisher

@Controller
open class EchoController: EchoApi {

    override fun postEcho(body: @Valid Publisher<Echo>): Publisher<Echo> {
        return body
    }
}
